client/stats/host.go

```diff
@@ -80,6 +83,8 @@ func (h *HostStatsCollector) Collect() (*HostStats, error) {
 			Free:      memStats.Free,
 		}
 		hs.Memory = ms
+	} else {
```

Comment:
@nak3 Yeah this is ok, since even if we collect more stats but return an error we won't update the stats.